### DISCUSSION TITLE
When or how often are OSM changes applied by BRouter?

### DISCUSSION THREAD

**Discussion by WolfgangZ42** [devemux86/cruiser#91] [Q&A] [answered]
We had here a bridge renovation that lasted months. Since 17 days, the bridge is open again which was also corrected in the OSM ([https://www.openstreetmap.org/changeset/133285578](https://www.openstreetmap.org/changeset/133285578)). While in Kurviger 2 the route guidance takes this into account is in Cruiser (BRouter) the bridge obviously still closed:
 
<img width="200" src="https://user-images.githubusercontent.com/121806238/226621119-66b95143-53de-4e42-a41a-36878e67c613.png"/>   <img width="200" src="https://user-images.githubusercontent.com/121806238/226621148-4aa98189-fb54-48b8-b0c9-dbea8d693b1c.png"/>   

No complaint - just a question!

**Comment by devemux86** [comment, marked as answer]
No problem to answer any question!

EDIT: _**they are  updated daily**_.
The [rd5 files](https://brouter.de/brouter/segments4/) have today's date.

@abrensch certainly can answer better such BRouter questions.

Have you updated the routing data with the BRouter app?
Have you tested on the [BRouter website](https://brouter.de/brouter-web/)?

**Reply by 0709wiwiwi** [reply]
The Bikerouter.de is updated more frequently. 
_Die Routingdaten werden aus [OpenStreetMap](https://www.openstreetmap.org/) viermal täglich neu erstellt. Das letzte Aktualisierungsdatum kann an den [Datendateien](https://brouter.m11n.de/segments/) abgelesen werden (Zeitstempel in UTC)._

https://marcusjaschen.com/

**Reply by devemux86** [reply]
I remember something similar about BRouter.
Probably their docs have not been updated?

**Reply by abrensch** [reply]
> No problem to answer any question!
> 
> I thought they are updated more often, like daily? The [rd5 files](https://brouter.de/brouter/segments4/) have today's date.
> 
> @abrensch certainly can answer better such BRouter questions.

Yes, daily. Refresh starts at night 1:03 AM and is available on the server around 6:30 AM. The timestamp on the files is the "map-snapshot time", so the time until which changes to the map are considered.

Marcus/bikerouter updates more often, i think just in a loop, as fast as it goes, without a defined daily cycle.

regards, Arndt

**Comment by MotoUKRider** [comment]
I decided to be nosey, so downloaded that area in BRouter and tried it - I got the short route!

![bridge-route](https://user-images.githubusercontent.com/125696109/226632915-d0efbedd-ee15-4d25-9557-de368a462a79.jpg)

**Reply by devemux86** [reply]
It is important to download and update the latest routing data.
And what BRouter profile and options are used in the routing.

**Comment by WolfgangZ42** [comment]
> It is important to download and update the latest routing data.

I'm not really familiar with the BRouter app - so far I've only used it to define profiles. I didn't see any map in BRouter ... 
How can I renew the routing data? 
BTW: in the BRouter website it works.

**Reply by 0709wiwiwi** [reply]
Brouter app.
 Go to download manager.
> confirm I know > see the map with update x (all) segments (or select individual)
Than wait until finished (all map segments blue)

**Reply by devemux86** [reply]
> How can I renew the routing data?

See also these BRouter images (similar process for updates):

- https://github.com/devemux86/cruiser/discussions/4#discussioncomment-4533509

**Reply by WolfgangZ42** [reply]
Sure I know this panel:
<img width="400" src="https://user-images.githubusercontent.com/121806238/226645469-8534a2c4-0877-4950-93aa-2eb979126b7b.png"/>  

But I was not aware, that I've to update these segments from time to time ...

**Reply by WolfgangZ42** [reply]
What should I say: after the update everything is fine! 
Again what learned - thanks for your help.
